Rakefile

```diff
@@ -237,6 +238,14 @@ def install_npm
   end
 end
 
+def ensure_code_studio_package
+  # never download if we build our own
+  return if CDO.use_my_code_studio
+
+  packager = S3Packaging.new('code-studio', code_studio_dir, dashboard_dir('public/code-studio-package'))
+  raise "No valid package found" unless packager.update_from_s3
```

Comment:
Style nit: I don't think a function call with side effects should be in a postfix unless clause.

I would find this more readable across two like this like this:
package_found = package.update_from_s3
raise "No valid package found" unless package_found